### DISCUSSION TITLE
To Do App

### DISCUSSION THREAD

**Discussion by eeriemyxi** [Textualize/textual#4116] [Show and tell]
https://github.com/Textualize/textual/assets/86738916/5e9144ad-9d05-42aa-b571-2a840a4dd713

I made a to-do app.

The source code for this is private frankly because the code is embarrassing: it is rushed and the entire code is a single five hundred line file, and I find that painful because if I wasn't forced to rush, I'd have spent more time for a much better architecture. I built this to get organized in life as soon as possible.  Other free to-do apps sucked, so I had to make a little one myself that suits my current needs. 

I was originally using Microsoft To Do, but it sucked. I liked how it auto detects when to notify, though; so I brought that feature to my app as well. If you type something like "[text] in 5 min*/hour*/sec*" it will detect it automatically if you want it to. Also I didn't like having to use my mouse to do stuff in Microsoft To Do, so I made mine completely dependent on keyboard.

I might re-implement this project next year, and if I do that, it's gonna be open source and nicely written.

**Comment by davep** [comment]
Looks really good; I'm liking the command-oriented side of it.

**Comment by diegorbb** [comment]
This looks dope btw. I would totally use it when it's open source. Any ideas if it'll be cross platform? Work is on macOS and personal machine is Windows/Linux.

**Reply by eeriemyxi** [reply]
Thanks for the review. And yes, even the current code is cross-platform. The library I am using for the notifications is cross-platform. The database currently is just a JSON file, but I will use `sqlite3` module (which is cross-platform) if I do work on this project again.

**Reply by diegorbb** [reply]
Thanks for letting me know mate. I really hope you do work on it again :) - good work 👏

**Comment by nnako** [comment]
Hi @eeriemyxi , unfortunately, I am not able to view the video within this post. So, although I am interested in tools for handling todo items, I can not check it out. Have you decided on open-sourcing your work? Would love to see it.


EDIT:

I managed to watch the video :-) . Looking forward to the code... ;-)

**Comment by Zaloog** [comment]
HI @eeriemyxi,
Nice work, I really like the timer/repeatability/notification feature. what did you use to implement it? Does it require the app to stay open?
And do not hesitate to share your code... we all started at some point

FYI, there is also [kanban-tui](https://github.com/Zaloog/kanban-tui), which I am maintaining.
It just got a lot of new features lately, like task dependencies + CLI interface to be useable by agents, and supports column/board customization. Backend is also sqlite and usability is keyboard focused

I might consider the timer feature also for kanban-tui.
